translation chapter 6 finished (#10)
iOiurson/svelte-fr-docs: svelte/fr/docs/06-accessibility-warnings.md

### `a11y-accesskey`

> Assurez-vous de ne pas utiliser l'attribut `accesskey` sur des éléments. L'attribut HTML `accesskey` permet aux développeurs web d'attribuer des raccourcis clavier aux éléments. Les incohérences entre les raccourcis clavier et les commandes clavier utilisées par le lecteur d'écran et les utilisateurs du clavier créent des complications d'accessibilité. Pour éviter les complications, les touches d'accès ne doivent pas être utilisées.

Reviewer: ```suggestion
Assure de ne pas utiliser l'attribut `accesskey` sur des éléments. L'attribut HTML `accesskey` permet aux développeurs web d'attribuer des raccourcis clavier aux éléments. Les incohérences entre les raccourcis clavier et les commandes clavier utilisées par le lecteur d'écran et les utilisateurs du clavier créent des complications d'accessibilité. Pour éviter les complications, les touches d'accès ne doivent pas être utilisées.
```

### `a11y-click-events-have-key-events`

> Assurez-vous que `on:click` soit accompagné d'au moins l'un des éléments suivants: `onKeyUp`, `onKeyDown`, `onKeyPress`. Penser à l'usage au clavier est important pour les utilisateurs avec des handicaps physiques qui ne peuvent pas utiliser de souris, pour les utilisateurs de liseuses d'écran, ainsi que pour la compatibilité AT.

Reviewer: ```suggestion
Assure que `on:click` soit accompagné d'au moins l'un des éléments suivants: `onKeyUp`, `onKeyDown`, `onKeyPress`. Penser à l'usage au clavier est important pour les utilisateurs avec des handicaps physiques qui ne peuvent pas utiliser de souris, pour les utilisateurs de liseuses d'écran, ainsi que pour la compatibilité AT.
```

### `a11y-distracting-elements`

> Assurez-vous qu'aucun élément distrayant ne soit utilisé. Les éléments distrayants visuellement peuvent causer des problèmes d'accessibilité avec les utilisateurs malvoyants. Ces éléments sont généralement dépréciés et doivent être évités.

Reviewer: ```suggestion
Assure qu'aucun élément distrayant ne soit utilisé. Les éléments distrayants visuellement peuvent causer des problèmes d'accessibilité avec les utilisateurs malvoyants. Ces éléments sont généralement dépréciés et doivent être évités.
```

### `a11y-img-redundant-alt`

> Assurez-vous que l'attribut `alt` des balises `img` ne contienne pas le mot "image" ou "photo". Les lecteurs d'écran décrivent déjà les éléments `img` comme étant des images. Il n'est pas nécessaire d'utiliser des mots tels que *photo* et / ou *image*.

Reviewer: ```suggestion
Assure que l'attribut `alt` des balises `img` ne contienne pas le mot "image" ou "photo". Les lecteurs d'écran décrivent déjà les éléments `img` comme étant des images. Il n'est pas nécessaire d'utiliser des mots tels que *photo* et / ou *image*.
```

> <img src="foo" alt="Foo mange un sandwich." />

Reviewer: Doublon (déjà présent en-dessous)

### `a11y-incorrect-aria-attribute-type`

> Assurez-vous que le bon type de valeur soit utilisé pour les attributs `aria`. Par exemple, `aria-hidden` ne devrait recevoir qu'un booléen.

Reviewer: ```suggestion
Assure que le bon type de valeur soit utilisé pour les attributs `aria`. Par exemple, `aria-hidden` ne devrait recevoir qu'un booléen.
```

```sv
<!-- A11y: la valeur de 'aria-hidden' doit être exactement `true` ou `false` -->
<div aria-hidden="yes"/>
```

---

### `a11y-invalid-attribute`

> Assurez-vous que les attributs importants pour l'accessibilité aient une valeur valide. Par exemple, `href` ne devrait pas être vide, `'#'` ou `javascript:`.

Reviewer: ```suggestion
Assure que les attributs importants pour l'accessibilité aient une valeur valide. Par exemple, `href` ne devrait pas être vide, `'#'` ou `javascript:`.
```

```sv
<!-- A11y: '' n'est pas un attribut href valide -->

> <a href=''>invalid</a>

Reviewer: ```suggestion
<a href=''>invalide</a>
```

```

---

### `a11y-label-has-associated-control`

> Assurez-vous qu'un élément `label` ait une étiquette de texte et un contrôle associé.

Reviewer: ```suggestion
Assure qu'un élément `label` ait une étiquette de texte et un contrôle associé.
```

Il existe deux méthodes prises en charge pour associer une étiquette à un contrôle :

- Envelopper un contrôle dans un élément `label`.
- Ajouter `for` à une étiquette et lui attribuer l'ID d'un champ d'entrée sur la page.

```sv
<label for="id">B</label>

<label>C <input type="text" /></label>

<!-- A11y: Une étiquette de formulaire doit être associée à un contrôle. -->
<label>A</label>
```

---

### `a11y-media-has-caption`

> Fournir des sous-titres pour les médias est essentiel pour les utilisateurs sourds pour suivre. Les sous-titres devraient être une transcription ou une traduction du dialogue, des effets sonores, des indications musicales pertinentes et d'autres informations audio pertinentes. Ce n'est pas seulement important pour l'accessibilité, mais peut également être utile pour tous les utilisateurs dans le cas où les médias ne sont pas disponibles (similaire au texte `alt` sur une image lorsqu'une image ne peut pas être chargée).

Reviewer: ```suggestion
Fournir des sous-titres pour les médias est essentiel afin que les utilisateurs sourds puissent suivre. Les sous-titres devraient être une transcription ou une traduction du dialogue, des effets sonores, des indications musicales pertinentes et d'autres informations audio pertinentes. Ce n'est pas seulement important pour l'accessibilité, mais peut également être utile pour tous les utilisateurs dans le cas où les médias ne sont pas disponibles (similaire au texte `alt` sur une image lorsqu'une image ne peut pas être chargée).
```

Les sous-titres doivent contenir toutes les informations importantes et pertinentes pour comprendre les médias correspondants. Cela peut signifier que les sous-titres ne sont pas une correspondance 1:1 du dialogue dans le contenu média. Cependant, les sous-titres ne sont pas nécessaires pour les composants vidéo avec l'attribut `muet`.

> Assurez-vous

Reviewer: ```suggestion
```

```sv
<video><track kind="captions"/></video>

<audio muted></audio>

<!-- A11y: Les éléments média doivent avoir une <track kind=\"captions\"> -->
<video></video>

<!-- A11y: Les éléments média doivent avoir une <track kind=\"captions\"> -->
<video><track /></video>
```

---

### `a11y-misplaced-role`

Certains éléments DOM réservés ne prennent pas en charge les rôles, états et propriétés ARIA. Cela est souvent dû à leur invisibilité, par exemple `meta`, `html`, `script`, `style`. Cette règle impose que ces éléments DOM ne contiennent pas l'attribut `role`.

```sv
<!-- A11y: <meta> ne devrait pas avoir l'attribut role -->
<meta role="tooltip">
```

---

### `a11y-misplaced-scope`

L'attribut `scope` ne devrait être utilisé que sur les éléments `<th>`.

```sv
<!-- A11y: L'attribut `scope` doit seulement être utilisé avec les élements <th> -->
<div scope="row" />
```

---

### `a11y-missing-attribute`

> Assurez-vous que les attributs requis l'accessibilité soient présents sur un élément. Cela inclut les vérifications suivantes:

Reviewer: ```suggestion
Assure que les attributs requis pour l'accessibilité soient présents sur un élément. Cela inclut les vérifications suivantes:
```

> - `<a>` devrait avoir un href (sauf s'il s'agit d'un [fragment-defining tag](https://github.com/sveltejs/svelte/issues/4697))

Reviewer: ```suggestion
- `<a>` devrait avoir un href (sauf s'il s'agit d'une [balise définissant un fragment](https://github.com/sveltejs/svelte/issues/4697))
```

- `<area>` devrait avoir alt, aria-label ou aria-labelledby
- `<html>` devrait avoir lang
- `<iframe>` devrait avoir title
- `<img>` devrait avoir alt
- `<object>` devrait avoir title, aria-label ou aria-labelledby
- `<input type="image">` devrait avoir alt, aria-label ou aria-labelledby

```sv
<!-- A11y: L'élément <input type=\"image\"> devrait avoir un attribut alt, aria-label ou aria-labelledby -->
<input type="image">

<!-- A11y: <html> L'élément <html> devrait avoir un attribut lang -->
<html></html>

<!-- A11y: L'élément <a> devrait avoir un attribut href -->
<a>text</a>
```

---

### `a11y-missing-content`

> Assurez-vous que les éléments d'en-tête (`h1`, `h2`, etc.) et les ancres aient un contenu et que ce contenu soit accessible aux lecteurs d'écran.

Reviewer: ```suggestion
Assure que les éléments d'en-tête (`h1`, `h2`, etc.) et les ancres aient un contenu, et que ce contenu soit accessible aux lecteurs d'écran.
```

```sv
<!-- A11y: L'élément <a> doit avoir un contenu enfant -->
<a href='/foo'></a>

<!-- A11y: L'élément <h1> doit avoir un contenu enfant -->
<h1></h1>
```

---

### `a11y-mouse-events-have-key-events`

> Assurez-vous que `on:mouseover` and `on:mouseout` soient accompagnés de `on:focus` et `on:blur`, respectivement. Cela aide à garantir que toutes les fonctionnalités déclenchées par ces événements de souris sont également accessibles aux utilisateurs du clavier.

Reviewer: ```suggestion
Assure que `on:mouseover` and `on:mouseout` soient accompagnés de `on:focus` et `on:blur`, respectivement. Cela aide à garantir que toutes les fonctionnalités déclenchées par ces événements de souris soient également accessibles aux utilisateurs du clavier.
```

```sv
<!-- A11y: on:mouseover doit être accompagné de on:focus -->
<div on:mouseover={handleMouseover} />

<!-- A11y: on:mouseout doit être accompagné de on:blur -->
<div on:mouseout={handleMouseout} />
```

---

### `a11y-no-redundant-roles`

Certains éléments HTML ont des rôles ARIA par défaut. Donner à ces éléments un rôle ARIA déjà défini par le navigateur n'a aucun effet et est redondant.

```sv
<!-- A11y: rôle redondant 'button' -->
<button role="button" />

<!-- A11y: rôle redondant 'img' -->
<img role="img" src="foo.jpg" />
```

---

### `a11y-no-interactive-element-to-noninteractive-role`

Les rôles [WAI-ARIA](https://www.w3.org/TR/wai-aria-1.1/#usage_intro) ne devraient pas être utilisés pour convertir un élément interactif en un élément non interactif. Les rôles ARIA non interactifs incluent `article`, `banner`, `complementary`, `img`, `listitem`, `main`, `region` et `tooltip`.

```sv
<!-- A11y: <textarea> ne peut pas avoir le rôle 'listitem' -->
<textarea role="listitem" />
```

---

### `a11y-no-noninteractive-tabindex`

> La navigation à l'aide de la touche Tab doit être limitée aux éléments de la page qui peuvent être interagis avec.

Reviewer: ```suggestion
La navigation à l'aide de la touche Tab doit être limitée aux éléments de la page avec lesquels il est possible d'interagir.
```

```sv
<!-- A11y: un élément non interactif ne peut pas avoir une valeur de tabIndex non négative -->
<div tabindex='0' />
```

---

### `a11y-positive-tabindex`

> Evitez les valeurs positives pour la propriété `tabindex`. Cela déplacera des éléments en dehors de l'ordre de tabulation attendu, ce qui créera une expérience confuse pour les utilisateurs du clavier..

Reviewer: ```suggestion
Évitez les valeurs positives pour la propriété `tabindex`. Cela positionnera des éléments en dehors de l'ordre de tabulation attendu, ce qui créera une expérience confuse pour les utilisateurs du clavier.
```

```sv
<!-- A11y: évitez les valeurs de tabindex supérieures à zéro -->
<div tabindex='1'/>
```

---

### `a11y-role-has-required-aria-props`

Les éléments avec des rôles ARIA doivent avoir tous les attributs requis pour ce rôle.

```sv
<!-- A11y: les éléments avec le rôle ARIA "checkbox" doivent avoir les attributs suivants définis: "aria-checked" -->
<span role="checkbox" aria-labelledby="foo" tabindex="0"></span>
```

---

### `a11y-structure`

> Assurez-vous que certains éléments DOM aient la bonne structure.

Reviewer: ```suggestion
Assure que certains éléments DOM aient la bonne structure.
```

```sv
<!-- A11y: <figcaption> doit être un enfant immédiat de <figure> -->
<div>
	<figcaption>Légende de l'image</figcaption>
</div>
```

---

### `a11y-unknown-aria-attribute`

> Assurez-vous que seuls les attributs ARIA connus soient utilisés. Cela est basé sur la spécification [WAI-ARIA States and Properties](https://www.w3.org/WAI/PF/aria-1.1/states_and_properties).

Reviewer: ```suggestion
Assure que seuls les attributs ARIA connus soient utilisés. Cela est basé sur la spécification [WAI-ARIA States and Properties](https://www.w3.org/WAI/PF/aria-1.1/states_and_properties).
```

```sv
<!-- A11y: Attribut aria inconnu 'aria-labeledby' (vouliez-vous dire 'labelledby'?) -->
<input type="image" aria-labeledby="foo">
```

---

### `a11y-unknown-role`

Les éléments avec des rôles ARIA doivent utiliser un rôle ARIA valide et non abstrait. Une référence aux définitions de rôle peut être trouvée sur le site [WAI-ARIA](https://www.w3.org/TR/wai-aria/#role_definitions).

```sv

> <!-- A11y: Rôle inconnu 'toooltip' (vouliez-vous dire 'tooltip'?) -->

Reviewer: De manière générale, je me rends que je ne sais pas si c'est une bonne idée de traduire tous ces commentaires, qui sont les warning affichés par le compilateur.

Je n'ai pas la réponse à ça...
